sdks/go/examples/timer_wordcap/wordcap.go

```diff
@@ -0,0 +1,228 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// timer_wordcap is a toy streaming pipeline that uses State and Timers with PubSub. It
+// does the following:
+//
+//	(1) create a topic and publish a few messages to it
+//	(2) Set user state and timer
+//
+// NOTE: it only runs on Dataflow and must be manually cancelled.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/graph/mtime"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/sdf"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/state"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/timers"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/io/rtrackers/offsetrange"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/log"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/register"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/x/beamx"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/x/debug"
+	"golang.org/x/exp/slog"
+)
+
+var (
+	input = flag.String("input", os.ExpandEnv("$USER-wordcap"), "Pubsub input topic.")
+)
+
+var (
+	data = []string{
+		"foo",
+		"bar",
+		"baz",
+	}
+)
+
+type Stateful struct {
+	ElementBag state.Bag[string]
+	TimerTime  state.Value[int64]
+	MinTime    state.Combining[int64, int64, int64]
+
+	OutputState timers.ProcessingTime
+}
+
+func NewStateful() *Stateful {
+	return &Stateful{
+		ElementBag: state.MakeBagState[string]("elementBag"),
+		TimerTime:  state.MakeValueState[int64]("timerTime"),
+		MinTime: state.MakeCombiningState[int64, int64, int64]("minTiInBag", func(a, b int64) int64 {
+			if a < b {
+				return a
+			}
+			return b
+		}),
+
+		OutputState: timers.InProcessingTime("outputState"),
+	}
+}
+
+func (s *Stateful) OnTimer(ctx context.Context, ts beam.EventTime, tp timers.Provider, key, timerKey, timerTag string, emit func(string, string)) {
+	switch timerKey {
+	case "outputState":
+		log.Infof(ctx, "Timer outputState fired on stateful for element: %v.", key)
+		s.OutputState.Clear(tp)
+		switch timerTag {
+		case "001":
+			log.Infof(ctx, "Timer with tag 001 fired on outputState stateful DoFn.")
+			s.OutputState.Set(tp, mtime.Now().ToTime().Add(1*time.Minute), timers.WithTag(timerTag))
+			emit(timerKey, timerTag)
+		}
+	}
+}
+
+func (s *Stateful) ProcessElement(ctx context.Context, ts beam.EventTime, sp state.Provider, tp timers.Provider, key, word string, emit func(string, string)) error {
+	s.ElementBag.Add(sp, word)
+	s.MinTime.Add(sp, int64(ts))
+
+	toFire, ok, err := s.TimerTime.Read(sp)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		toFire = int64(mtime.Now().Add(1 * time.Minute))
+	}
+	minTime, _, err := s.MinTime.Read(sp)
+	if err != nil {
+		return err
+	}
+
+	s.OutputState.Set(tp, mtime.Time(toFire).ToTime(), timers.WithOutputTimestamp(mtime.Time(minTime).ToTime()), timers.WithTag(word))
```

Comment:
IIRC, we have both minTime and minTime in millis? We should just call https://pkg.go.dev/time#UnixMilli here instead.

Ideally we'd just store time.Times directly in the state, but that feels like a different bug to fix if it doesn't already work.